src/main/java/database/PostsRepository.java

```diff
@@ -42,17 +51,120 @@ public Post create(Post post) {
     }
 
     @Override
-    public Post read(String id) {
-        return null;
+    public Post read(Integer id) {
+
+        Post post = null;
+
+        try {
+            Statement stmt = connection.createStatement();
+
+            ResultSet rs = stmt.executeQuery(String.format(querySelectPostById, id));
+
+            if(rs.next()) {
+
+                post = convertToPost(rs);
+            }
+
+            stmt.close();
+        } catch(Exception ignored) {
+
+            return null;
+        }
+
+        return post;
+    }
+
+    private Post convertToPost(ResultSet rs) throws SQLException {
+
+        Integer id = rs.getInt("id");
+        Integer userId = rs.getInt("user_id");
+        Boolean hasForm = rs.getBoolean("has_form");
+        Status status = Status.valueOf(rs.getString("status"));
+
+        return new Post.PostBuilder(id, userId).hasForm(hasForm).withStatus(status).build();
     }
 
     @Override
     public Post update(Post object) {
+        // TODO: Should we change the signature with one that contains
```

Comment:
That, or we can add the id of the original. I don't really know what option is the best one